src/plugins/discover/public/application/angular/discover.js

```diff
@@ -291,6 +298,31 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     }
   });
 
+  data.search.session.setSearchSessionRestorationInfoProvider({
+    getName: async () => 'Discover',
+    getUrlGeneratorData: async () => {
```

Comment:
Will review this today, initial thought, I'm trying the move stuff out of `discover.js` to deangularize and typescriptify. so since this is using lot's of appState, could we move this function there? so you could use `buildUrlGenerateData` function with arguments data (for filterManager, sessionId, timefilter), savedSearch. wdyt?

Reply:
Moved 👍